### PR TITLE
Optimize idxmin, idxmax with dask

### LINKED ISSUE #9425: idxmin / idxmax is not parallel friendly
### Discussed in https://github.com/pydata/xarray/discussions/9421

<div type='discussions-op-text'>

<sup>Originally posted by **KBodolai** September  2, 2024</sup>
Hi there! I have a question about the chunking behaviour when using idxmin / idxmax for a chunked array.

What is the expected behaviour for the chunks after we run `idxmin` over one of the dimensions?  Naively I'd expect it to keep the chunks along the other dimensions, but that doesn't seem to be what happens: (Example below with time, x, y)

```python
import numpy as np
import xarray as xr

# create some dummy data and chunk
x, y, t = 1000, 1000, 57
rang = np.arange(t*x*y)
da = xr.DataArray(rang.reshape(t, x, y), coords={'time':range(t), 'x': range(x), 'y':range(y)})
da = da.chunk(dict(time=-1, x=256, y=256))
```

Now when I look at the array, it looks something like this:

Screenshot 2024-09-02 at 17 06 22

```python
da.idxmin('time')
```

But after doing `idxmin` I get the outputs below
Screenshot 2024-09-02 at 17 00 17

My understanding is that it seems to be trying to keep the number of chunks. But oddly, when we do it for floats:

```python
da = da.astype('float32')
```

before and after doing the `idxmin` looks like this:

Screenshot 2024-09-02 at 17 10 25
Screenshot 2024-09-02 at 17 10 11

Is this the expected behavour for this operation? I'm guessing the reshaping in the source code happens [here](https://github.com/pydata/xarray/blob/12d8cfa317b1a08748cb6a0a1b10a6efe3a7d3d2/xarray/core/computation.py#L2180), but I haven't been able to figure out how yet.

Thanks!
K.
</div>

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Closes #9425
- [x] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

cc @phofl here we need to index a numpy array with a dask array (commonly a much larger array) in a sane manner.

We now preserve chunksizes for
```
import numpy as np
import xarray as xr

# create some dummy data and chunk
x, y, t = 1000, 1000, 57
rang = np.arange(t*x*y)
da = xr.DataArray(rang.reshape(t, x, y), coords={'time':range(t), 'x': range(x), 'y':range(y)})
da = da.chunk(dict(time=-1, x=256, y=256))
da.idxmin('time')
```

***After***
image

***Before***
image

### CODE REVIEW

File: xarray/tests/test_dataarray.py [hunk 1]
```diff
@@ -4949,7 +4949,15 @@ def test_argmax(
 
         assert_identical(result2, expected2)
 
-    @pytest.mark.parametrize("use_dask", [True, False])
+    @pytest.mark.parametrize(
+        "use_dask",
+        [
+            pytest.param(
+                True, marks=pytest.mark.skipif(not has_dask, reason="no dask")
```
Comment (headtr1ck) [inline xarray/tests/test_dataarray.py:4956]:
```suggestion
                True, marks=requires_dask
```

**Comment by dcherian** [comment]
Merging on Thursday if there are no comments

**Review by headtr1ck — COMMENTED** [review]
Ups, never submitted my review comments.

File: xarray/tests/test_dask.py [hunk 2]
```diff
@@ -1814,3 +1814,16 @@ def test_minimize_graph_size():
         # all the other dimensions.
         # e.g. previously for 'x',  actual == numchunks['y'] * numchunks['z']
         assert actual == numchunks[var], (actual, numchunks[var])
+
+
+def test_idxmin_chunking():
```
[the remark this answers is not in the document]
Comment (dcherian) [reply xarray/tests/test_dask.py:1819]: the whole file has an importorskip("dask")

**Pull request merged**